model_compression_toolkit/core/common/similarity_analyzer.py

```diff
@@ -51,19 +51,24 @@ def _similarity_tensor_norm(x: np.ndarray, p: float = 2.0) -> np.ndarray:
     return (np.abs(x) ** p).sum(axis=-1) ** (1.0/p)
 
 
-def flatten_tensor(t: np.ndarray, batch: bool) -> np.ndarray:
+def flatten_tensor(t: np.ndarray, batch: bool, axis: int = None) -> np.ndarray:
     """
     Flattening the samples batch to allow similarity analysis computation per sample.
 
     Args:
         t: A tensor to be flattened.
         batch: Whether the similarity computation is per image or per tensor.
+        axis: Axis along which the operator has been computed.
 
     Returns: A flattened tensor which has the number of samples as is first dimension.
 
     """
 
-    if batch:
+    if axis is not None and batch:
+        f_t = t.reshape([t.shape[0], -1, t.shape[axis]])
```

Comment:
this only works when "axis" is the last axis. need to permute the "axis" to the last index before reshaping like that